danny-avila/LibreChat
# 🤲 feat(a11y): Initial broad sweep of a11y, added linters, tests; fix: close sidebars in mobile view

Closes #3474 
Closes https://github.com/danny-avila/LibreChat/issues/3534 (unrelated to a11y)

Partially addresses #3458
Addresses #3476, #3514, #3515

I implemented various accessibility improvements across the application and added a11y testing to our CI pipeline. These changes enhance the user experience for those using assistive technologies and ensure our application adheres to web accessibility standards, addressing several reported issues.

- Added a GitHub Actions workflow for accessibility linting, including the client/src/** path.
- Implemented visible focus outlines for various UI components such as buttons, menus, and form controls.
- Added appropriate ARIA labels and roles to improve screen reader compatibility.
- Refactored components to use semantic HTML elements where appropriate, enhancing the overall document structure.
- Improved color contrast for better visibility, particularly in the MessageAudio component.
- Added a heading structure to the Landing page to aid in content organization.
- Implemented keyboard navigation support for interactive elements.
- Updated the lang attribute in index.html to "en-US" for proper language specification.
- Set up Playwright configuration for a11y testing and created initial a11y test specs.
- Refactored the ChatRoute component to include an accessible loading spinner with a live region.
- Updated various components to resolve ESLint warnings related to accessibility.
- Fixed the scroll to bottom button missing an accessible name (Closes #3474).
- Addressed landmark a11y issues (addresses #3514, #3515).

### Before (Landing)

![image](https://github.com/user-attachments/assets/7aa57e54-3f9c-4105-bd8c-3a3a9e2fc859)

### After (Landing)

![image](https://github.com/user-attachments/assets/e6667814-5ea6-44f6-b648-b0d144edb3ab)

## Testing

To test these changes:

1. Run the application locally and navigate through the UI using only a keyboard to ensure all interactive elements are accessible.
2. Use a screen reader to navigate the application and verify that all content is properly announced.
3. Run the new a11y tests using Playwright to ensure no accessibility issues are detected.
4. Verify that the GitHub Actions workflow for accessibility linting passes successfully.
5. Specifically test the scroll to bottom button functionality with a screen reader to confirm it now has an accessible name.
6. Check that the main content area is properly identified by assistive technologies.


## Checklist
